### ISSUE TITLE
Reporting incorrect line for ill-formed escape sequence

### ISSUE THREAD

**Issue by ericcornelissen** [htmllint/htmllint#233]
I found an issue with htmllint through a plugin for Atom ([#1](https://github.com/AtomLinter/linter-htmllint/issues/1)) where the incorrect line is reported, but the column is correct.

The following HTML can be used to reproduce the issue:

```html
 1. <!DOCTYPE html>
 2. <html>
 3.     <head>
 4.     </head>
 5.     <body>
 6.         <div>
 7.             <h5>Star Wars</h5>
 8.             <p>These are not the droids & you're looking for</p>
 9.         </div>
10.     </body>
11. </html>
```

I used the following script to verify the issue comes from htmllint itself. As you can see it says the second issue is on line 9 while it should be 8 (where `test.html` contains the html shown above):

```js
let fs = require('fs')
let lint = require('htmllint');

let file = fs.readFileSync('test.html', 'utf8')
let result = lint(file);

result.then(console.log);

// [ Issue {
//     line: 3,
//     column: 5,
//     code: 'E027',
//     data: {},
//     rule: 'head-req-title' },
//   Issue {
//     line: 9,
//     column: 42,
//     code: 'E023',
//     data:
//      { chars: '& ',
//        part: 'text',
//        desc: 'an ill-formed escape sequence' },
//     rule: 'spec-char-escape' } ]
```

**Comment by mlochbaum** [comment]
Thanks for the report. That rule ended up converting to 1-based indices twice, so that all line and column numbers were one too large. Some day I'll start adding line numbers to our unit tests...

**Comment by ericcornelissen** [comment]
Fix works as expected, thank you 🎉
